wow/core/src/main/scala/wow/realm/handlers/MovementHandler.scala

```diff
@@ -2,60 +2,66 @@ package wow.realm.handlers
 
 import wow.Application
 import wow.realm.events.PlayerMoved
+import wow.realm.protocol._
 import wow.realm.protocol.payloads.ClientMovement
-import wow.realm.protocol.{MultiPayloadHandlerFactory, OpCodes, PacketSerialization, PayloadHandler}
+import wow.realm.session.NetworkWorker
 
 /**
   * Movement packet handler
+  *
+  * Uses OpCodeProvider.None as OpCodes are overloaded below
   */
-class MovementHandler extends PayloadHandler[ClientMovement] {
-  private val eventStream = context.system.eventStream
-
+object MovementHandler
+  extends PayloadHandler[NetworkWorker, ClientMovement](MovementHandlerCompanion.SupportedOpCodes) {
   /**
     * Processes an incoming payload
     *
     * @param payload payload to be processed
     */
-  override protected def process(payload: ClientMovement): Unit = {
+  override protected def handle(header: ClientHeader, payload: ClientMovement)(self: NetworkWorker): Unit = {
+    import self._
+
     val clientDelay = Application.uptimeMillis() - payload.time
 
     val adjustedPayload = payload.copy(time = payload.time + clientDelay)
 
-    val (headerBits, payloadBits) = PacketSerialization.outgoingSplit(adjustedPayload, currentOpCode)
+    val (headerBits, payloadBits) = PacketSerialization.outgoingSplit(adjustedPayload, header.opCode)
 
-    eventStream.publish(PlayerMoved(payload, headerBits, payloadBits))
+    realm.eventStream.publish(PlayerMoved(payload, headerBits, payloadBits))
 
-    log.debug(s"Player ${payload.guid.id} moved to ${payload.position} with $currentOpCode")
+    log.debug(s"Player ${payload.guid.id} moved to ${payload.position} with ${header.opCode}")
   }
 }
 
-object MovementHandler extends MultiPayloadHandlerFactory[MovementHandler, ClientMovement](
-  OpCodes.MsgMoveStartForward,
-  OpCodes.MsgMoveStartBackward,
-  OpCodes.MsgMoveStop,
-  OpCodes.MsgMoveStartStrafeLeft,
-  OpCodes.MsgMoveStartStrafeRight,
-  OpCodes.MsgMoveStopStrafe,
-  OpCodes.MsgMoveJump,
-  OpCodes.MsgMoveStartTurnLeft,
-  OpCodes.MsgMoveStartTurnRight,
-  OpCodes.MsgMoveStopTurn,
-  OpCodes.MsgMoveStartPitchUp,
-  OpCodes.MsgMoveStartPitchDown,
-  OpCodes.MsgMoveStopPitch,
-  OpCodes.MsgMoveSetRunMode,
-  OpCodes.MsgMoveSetWalkMode,
-  OpCodes.MsgMoveFallLand,
-  OpCodes.MsgMoveStartSwim,
-  OpCodes.MsgMoveStopSwim,
-  OpCodes.MsgMoveSetFacing,
-  OpCodes.MsgMoveSetPitch,
-  OpCodes.MsgMoveHeartbeat,
-  OpCodes.MoveFallReset,
-  OpCodes.MoveSetFly,
-  OpCodes.MsgMoveStartAscend,
-  OpCodes.MsgMoveStopAscend,
-  OpCodes.MoveChangeTransport,
-  OpCodes.MsgMoveStartDescend
-)
+object MovementHandlerCompanion {
+  val SupportedOpCodes: OpCodes.ValueSet = OpCodes.ValueSet(
+    OpCodes.MsgMoveStartForward,
```

Comment:
Not sure what you mean, I would need to import OpCodes._ for that to work no ? If that's what you're suggesting, I agree.